napari/components/_viewer_key_bindings.py

```diff
@@ -18,18 +26,14 @@ def reset_scroll_progress(viewer):
 reset_scroll_progress.__doc__ = trans._("Reset dims scroll progress")
 
 
-@ViewerModel.bind_key('Control-Y')
+@ViewerModel.register_action(description=trans._("Toggle ndisplay."))
```

Comment:
why did the actual key binding have to be moved into the `_actions` dict?  I don't love the pattern that we seem to now be creating "action ids" as _strings_ that are important and meaningful, but spread across a few different places in the codebase.  (maybe it was already that way and I didn't realize it?) 
Couldn't `KeymapProvider.register_action` also have a `shortcut` parameter?

Reply:
>Couldn't KeymapProvider.register_action also have a shortcut parameter?

We will need to have shortcuts centralized, since they will be provided by language packs (eventually), so having shortcuts defined everywhere in the codebase "manually" feels a bit odd.

Reply:
I get that.  But I think it feels equally odd to have a callback out there somewhere that is only matched by nature of using the same name ...

Reply:
>I get that. But I think it feels equally odd to have a callback out there somewhere that is only matched by nature of using the same name ...

You mean using the name of the function. would something like this be better:

```python
class ViewerActions:
     action_name = "some_unique_indetifier"

@ViewerModel.register_action(id_=ViewerActions.action_name, description=trans._("Toggle ndisplay."))
...
```
?

Reply:
I guess I just want all of the action registration in one place. So rather than using a decorator to register a function somewhere, and another function somewhere else to register the shortcut for that same action, id prefer the shortcut registration and callback binding to be in the same place if possible (even if that means we don’t decorate the function anymore)

Reply:
That sort of conflict with configurability. Because your decorator/register_and_bind, may not bind depending on user preferences.

Reply:
I would strongly prefer user defined names, I stay far away from using qualnames, as it make all the filenames, and classnames part of the API; keep in mind those names will be used for configuration and stored in configuration files.

we could try to use `package_name:action_name` if you wish, but I think decoupling name and fully qualified name make sens.

Reply:
For some context, In Spyder we defined that actions should be unique to a given plugin, which is effectively the same as the proposed `package_name:action_name`.

When plugins register actions, I guess we will already know "who" is doing the registration so I think `package_name:action_name` 👍🏼

Reply:
Yep, that seems good enough for me, although it's still a major source of non-locality within a big package like napari. But it's 100x better than just "name". 😊

Reply:
btw, I genuinely don't know the best answer to this: do we want `package_name` to be determined from the function's source, or manually? ie can scikit-learn register an action with the name `scikit:filter`? Cos that might upset scikit-image devs. 😂